datarepo-clienttests/src/main/java/testscripts/RetrieveDRSSnapshot.java

```diff
@@ -0,0 +1,216 @@
+    JobModel ingestTabularDataJobResponse =
+        repositoryApi.ingestDataset(datasetSummaryModel.getId(), ingestRequest);
+
+    ingestTabularDataJobResponse =
+        DataRepoUtils.waitForJobToFinish(repositoryApi, ingestTabularDataJobResponse);
+    IngestResponseModel ingestResponse =
+        DataRepoUtils.expectJobSuccess(
+            repositoryApi, ingestTabularDataJobResponse, IngestResponseModel.class);
+
+    System.out.println("successfully loaded data into dataset: " + datasetSummaryModel.getName());
+
+    // make the create snapshot request and wait for the job to finish
+    JobModel createSnapshotJobResponse =
+        DataRepoUtils.createSnapshot(
+            repositoryApi, datasetSummaryModel, "snapshot-simple.json", true);
+
+    // save a reference to the snapshot summary model so we can delete it in cleanup()
+    snapshotSummaryModel =
+        DataRepoUtils.expectJobSuccess(
+            repositoryApi, createSnapshotJobResponse, SnapshotSummaryModel.class);
+
+    System.out.println("successfully created snapshot: " + snapshotSummaryModel.getName());
+  }
+
+  public void userJourney(ApiClient apiClient) throws Exception {
+    RepositoryApi repositoryApi = new RepositoryApi(apiClient);
+    DataRepositoryServiceApi dataRepositoryServiceApi = new DataRepositoryServiceApi(apiClient);
+
+    SnapshotModel snapshotModel = repositoryApi.retrieveSnapshot(snapshotSummaryModel.getId());
+    TableModel tableModel = snapshotModel.getTables().get(0);
+    BigQuery bigQuery =
+        BigQueryOptions.newBuilder()
+            .setProjectId(snapshotModel.getDataProject())
+            .build()
+            .getService();
+    // System.out.println(snapshotModel.getDataProject()); // broad-jade-mm-data
+    // System.out.println(source.get(0).getDataset().getName()); //
+    // DatasetSimple13213724609325394569
+    // System.out.println(tableModel.getName()); // vcf_file
+    String queryForFileRefs =
+        "SELECT * FROM "
+            + snapshotModel.getDataProject()
+            + "."
+            + snapshotModel.getName()
+            + "."
+            + tableModel.getName();
+
+    QueryJobConfiguration queryConfig = QueryJobConfiguration.newBuilder(queryForFileRefs).build();
+    TableResult result = bigQuery.query(queryConfig);
+
+    ArrayList<String> fileRefs = new ArrayList<>();
+    result.iterateAll().forEach(r -> fileRefs.add(r.get("VCF_File_Ref").getStringValue()));
+    // fileRefs should only be 1 in size
+    String fileModelFileId = fileRefs.get(0);
+    String fileId = fileModelFileId.split("_")[2];
+    String dirObjectId = "v1_" + snapshotSummaryModel.getId() + "_" + fileId;
+    DRSObject object = dataRepositoryServiceApi.getObject(dirObjectId, false);
+
+    System.out.println(
+        "successfully retrieved drs object: "
+            + object.getName()
+            + " with id: "
+            + dirObjectId
+            + ", data project: "
+            + snapshotModel.getDataProject());
```

Comment:
I pulled it out---but take a look. I'm not sure I'm getting this exactly as you intended

Reply:
This is what I meant, except for the retrieveSnapshot lookup in the userJourney method. Can you use the SnapshotModel you retrieved in the setup method instead of re-retrieving it here?